app/src/main/java/com/example/android/sunshine/app/DetailFragment.java

```diff
@@ -193,37 +193,46 @@ public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
             mFriendlyDateView.setText(friendlyDateText);
             mDateView.setText(dateText);
 
-            // Read description from cursor and update view
-            String description = data.getString(COL_WEATHER_DESC);
+            // Get description from weather condition ID
+            String description = Utility.getStringForWeatherCondition(getActivity(), weatherId);
```

Comment:
It has, but I don't see what difference it makes. There is some logic behind how the weather description is got, and it is extracted in the `Utility` class now.